### PR TITLE
WooCommerce: Add actions, reducers, and selectors for fetching products settings

### PR DESCRIPTION
**Pull request by justinshreve**
This PR adds actions, reducers, and selectors for fetching products settings (`/settings/products`). It follows the same pattern the general settings uses. 

It includes two selectors for getting weight and dimensions units which are the main settings we need from the endpoint (for displaying the correct unit on the product form). Includes README.

To Test:
* Run `make test` and make sure all tests pass.

See #14185.

### CODE REVIEW

**Comment by justinshreve** [comment]
I have rebased and updated this to use the new action type names (#14832).

File: client/extensions/woocommerce/state/sites/settings/products/test/selectors.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		it( 'should get the weight unit setting from the state.', () => {
+			expect( getWeightUnitSetting( loadedState, 123 ) ).to.eql( weightUnitSetting );
+		} );
+
+		it( 'should get the siteId from the UI tree if not provided.', () => {
+			expect( getWeightUnitSetting( loadedStateWithUi ) ).to.eql( weightUnitSetting );
+		} );
+	} );
+
+	describe( '#getDimensionsUnitSetting,', () => {
```
Comment (ryelle) [inline client/extensions/woocommerce/state/sites/settings/products/test/selectors.js]: Stray comma in the label here

File: client/extensions/woocommerce/app/products/product-form-simple-card.js [hunk 2]
```diff
@@ -60,28 +60,21 @@ const ProductFormSimpleCard = ( { product, editProduct, translate } ) => {
...
 				<FormFieldSet className="products__product-dimensions-input">
 					<FormLabel>{ translate( 'Dimensions' ) }</FormLabel>
 					<FormDimensionsInput
-						unit={ translate( 'in' ) }
 						dimensions={ product.dimensions }
 						onChange={ setDimension }
 					/>
 				</FormFieldSet>
 				<FormFieldSet className="products__product-weight-input">
 					<FormLabel>{ translate( 'Weight' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/products/product-form-simple-card.js]: Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Weight', { context: 'Dns Record'} )` *ES Score: 19*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/woocommerce-product-units)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx [hunk 3]
```diff
@@ -1,56 +1,112 @@
...
+					name="width"
+					placeholder={ translate( 'W', { comment: 'Width placeholder for dimensions input' } ) }
+					type="number"
+					value={ dimensions && dimensions.width || '' }
+					onChange={ onChange }
+					className="form-dimensions-input__width"
+				/>
+				<FormTextInputWithAffixes
+					name="height"
+					placeholder={ translate( 'H', { comment: 'Height placeholder for dimensions input' } ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Hi! I've found a possible matching string that has already been translated **8** times:
`translate( 'h' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx [hunk 4]
```diff
@@ -1,56 +1,112 @@
...
+
+	render() {
+		const { className, noWrap, dimensions, onChange, translate, dimensionsUnit } = this.props;
+		const classes = classNames( 'form-dimensions-input', className, { 'no-wrap': noWrap } );
+
+		return (
+			<div className={ classes }>
+				<FormTextInput
+					name="length"
+					placeholder={ translate( 'L', { comment: 'Length placeholder for dimensions input' } ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Hi! I've found a possible matching string that has already been translated **34** times:
`translate( 'L', { context: 'momentjs format string (sameElse)'} )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**